functions.php

```diff
@@ -437,7 +438,12 @@ function getAlert($guid, $connection2, $gibbonAlertLevelID)
 function getSalt()
 {
     $c = explode(' ', '. / a A b B c C d D e E f F g G h H i I j J k K l L m M n N o O p P q Q r R s S t T u U v V w W x X y Y z Z 0 1 2 3 4 5 6 7 8 9');
-    $ks = array_rand($c, 22);
+
+    for ($x = 0; $x < 22; $x++) {
+        $ind =  mt_rand(0, count($c) - 1);
+        $ks = $c[$ind];
+    }
+
```

Comment:
In the original implementation, `$ks` is an array. It can then be used in the subsequence `foreach` no problem.

In the new implementation, `$ks` was only assigned to a single random character, which I don't think will give you the intended result. There are also other issue with it when running in PHP 8.2.